coalib/core/Core.py

```diff
@@ -339,14 +340,15 @@ def run(bears, result_callback):
 
             def result_callback(result):
                 pass
+    :param executor:
+        Custom executor used to run the bears. Must be subclass of
+        _base.Executor
     """
-    # FIXME Allow to pass different executors nicely, for example to execute
-    # FIXME   coala with less cores, or to schedule jobs on distributed systems
-    # FIXME   (for example Mesos).
 
     # Set up event loop and executor.
     event_loop = asyncio.SelectorEventLoop()
-    executor = concurrent.futures.ProcessPoolExecutor()
+    if executor is None or not issubclass(type(executor), _base.Executor):
```

Comment:
Yeah, you're right. Its programmer's duty to pass valid executor. Since its not directly used by end user, we can assume it'll be proper.